webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/FindDataService.scala

```diff
@@ -231,28 +237,51 @@ class FindDataService @Inject()(dataServicesHolder: BinaryDataServiceHolder)(imp
 
   def createHistogram(dataSource: DataSource, dataLayer: DataLayer) = {
 
-    def calculateHistogramValues(data: Array[_ >: UByte with UShort with UInt with ULong with Float]) = {
-      val counts = Array.ofDim[Long](256)
+    def calculateHistogramValues(data: Array[_ >: UByte with UShort with UInt with ULong with Float],
+                                 isUint24: Boolean = false) = {
+      val counts = if (isUint24) Array.ofDim[Long](768) else Array.ofDim[Long](256)
+      var extrema: (Option[Float], Option[Float]) = (None, None)
+
       data match {
-        case byteData: Array[UByte] =>
-          byteData.foreach(el => counts(el.toInt) += 1)
+        case byteData: Array[UByte] => {
+          if (!isUint24)
+            byteData.foreach(el => counts(el.toInt) += 1)
+          else {
+            for (i <- byteData.indices by 3) {
+              counts(byteData(i).toInt) += 1
+              counts(byteData(i + 1).toInt + 256) += 1
+              counts(byteData(i + 2).toInt + 512) += 1
+            }
+          }
+        }
         case shortData: Array[UShort] =>
           shortData.foreach(el => counts((el / UShort(256)).toInt) += 1)
         case intData: Array[UInt] =>
           intData.foreach(el => counts((el / UInt(16777216)).toInt) += 1)
         case longData: Array[ULong] =>
           longData.foreach(el => counts((el / ULong(math.pow(2, 56).toLong)).toInt) += 1)
         case floatData: Array[Float] =>
-          floatData.foreach(el => counts(Math.clamp(Math.roundDown(el * 255), 0, 255)) += 1)
+          val (min, max) = floatData.foldLeft((floatData(0), floatData(0))) {
+            case ((currMin, currMax), e) => (math.min(currMin, e), math.max(currMax, e))
+          }
+          val bucketSize = (max - min + 1) / 256
+          floatData.foreach(el => counts(Math.clamp(Math.roundDown((el - min) / bucketSize), 0, 255)) += 1)
+          extrema = (Some(min), Some(max))
+      }
+      if (!isUint24)
```

Comment:
got me confused here for a moment, I’d suggest you use the positive condition and swap the statements :)